Modules/people_options.rb

```diff
@@ -0,0 +1,57 @@
+require_relative '../Components/person'
+require_relative '../Entities/student'
+require_relative '../Entities/teacher'
+
+class PeopleOptions
+  attr_accessor :people_list
+
+  def initialize
+    @people_list = []
+  end
+
+  def list_all_people
+    if @people_list.empty?
+      puts 'No record found! Add a person...'
+    else
+      puts 'All people in the library'
+      @people_list.each do |person|
+        puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student(1) or a teacher(2)? [Input the number] '
+    person_role = gets.chomp
+    case person_role
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Please add a valid input!'
+    end
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase == 'y'
```

Comment:
- [ ] it would be better to separate the code that requests and validates inputs from the users like:
```ruby 
book_num = gets.chomp.to_i # code that requests for inputs 
 puts 'Select a person from the following list by number (not id)'
```
and put them in methods and then use the methods inside `create_rental`, `create_book`, `create_person` `list_book`...etc that way you will achieve the single responsibility responsible please try to do that for all the methods